frontend/app/src/connection/ConnectionManager.ts

```diff
@@ -136,16 +137,42 @@ export class ConnectionManager {
     }
   }
 
+  /**
+   * Checks query params to determine if static notebook Id has been passed.
+   * If so, returns the Id.
+   */
+  private checkStaticConnection(): string | null {
+    const queryParams = new URLSearchParams(document.location.search)
+    return queryParams.get("staticAppId")
+  }
+
+  /**
+   * Establish either a WebsocketConnection or StaticConnection
+   * based on query params.
+   */
   private async connect(): Promise<void> {
-    try {
-      this.connection = await this.connectToRunningServer()
-    } catch (e) {
-      const err = ensureError(e)
-      logError(err.message)
-      this.setConnectionState(
-        ConnectionState.DISCONNECTED_FOREVER,
-        err.message
-      )
+    const staticAppId = this.checkStaticConnection()
+
+    if (staticAppId) {
+      // Establish a static connection
+      establishStaticConnection({
+        staticAppId,
+        onConnectionStateChange: this.setConnectionState,
+        onMessage: this.props.onMessage,
+        onConnectionError: this.props.onConnectionError,
+      })
+    } else {
+      // Establish a websocket connection
+      try {
+        this.connection = await this.connectToRunningServer()
```

Comment:
Seems weird that `this.connection` is not used for the static connection. Wanted to offer a couple solutions.

1. Perhaps we should rename `this.connection` to `this.websocketConnection`? That way, it's explicit.
2. We modify the type to be null and _explicitly_ set it to null on static connection (with a comment explaining that saving a connection is unneccessary for XYZ reasons).